File: composer/src/main/java/io/piano/android/composer/model/ExperienceResponse.java
package io.piano.android.composer.model;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

@Keep
public class ExperienceResponse {
    @Nullable
    @SerializedName("tbc")
    public CookieObject tbCookie;

    @Nullable
    @SerializedName("xbc")
    public CookieObject xbCookie;

    @Nullable
    @SerializedName("tac")
    public CookieObject taCookie;

    @SerializedName("timezone_offset")
    public int timeZoneOffsetMillis;

    @Nullable
    @SerializedName("visit_timeout")
    public Integer visitTimeoutMinutes;

    @NonNull
    @SerializedName("result")
    public EventsContainer result;
}

File: composer/src/main/java/io/piano/android/composer/model/ExperienceResponse.kt
package io.piano.android.composer.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
public class ExperienceResponse(
    @JvmField
    @Json(name = "tbc")
    public val tbCookie: CookieObject?,
    @JvmField
    @Json(name = "xbc")
    public val xbCookie: CookieObject?,
    @JvmField
    @Json(name = "tac")
    public val taCookie: CookieObject?,
    @JvmField
    @Json(name = "timezone_offset")
    public val timeZoneOffsetMillis: Int,
    @JvmField
    @Json(name = "visit_timeout")
    public val visitTimeoutMinutes: Long?,
    @JvmField
    @Json(name = "bid")
    public val browserId: String?,
    @JvmField
    @Json(name = "uid")
    public val userId: String?,
    @JvmField public val cxenseCustomerPrefix: String?,
    @JvmField public val result: EventsContainer,
)
